### PR TITLE
#10 Add PercentageValidator and tests

### PR DESCRIPTION
**Pull request by Reisclef**

### CODE REVIEW

**Comment by afucher** [comment]
Hey @Reisclef thanks for your PR! 🎉 

One thing that I was expecting in a Percentage validator is that the user should be able to input with the percentage symbol, like "50%".

File: test/InquirerUnitTest/Validators/PercentageValidatorUnitTest.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+            isValid.Should().BeFalse();
+        }
+
+        [Fact]
+        public void ShouldReturnErrorMessage()
+        {
+            var validator = new PercentageValidator();
+
+            validator.GetErrorMessage().Should().Be("Sorry, a percentage must be a number between 0 and 100");
```
Comment (afucher) [inline test/InquirerUnitTest/Validators/PercentageValidatorUnitTest.cs]: We don't need to add "Sorry,"at the error message.

File: test/InquirerUnitTest/Validators/PercentageValidatorUnitTest.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public class PercentageValidatorUnitTest
+    {
+        [Theory]
+        [InlineData("1")]
+        [InlineData("50")]
+        [InlineData("100")]
+        [InlineData("0.01")]
+        [InlineData("0")]
+        public void ShouldAcceptValuesBetween0And100(string aPercentValue)
```
Comment (afucher) [inline test/InquirerUnitTest/Validators/PercentageValidatorUnitTest.cs:20]: Values with the percentage symbol should be valid, like 50%

**Review by afucher — CHANGES_REQUESTED** [review]
Thanks for your PR.

**Comment by Reisclef** [comment]
Okay, cool. I wasn't sure whether that would be the case, and I didn't want to assume. Have a look at the newest commit and let me know if anything else needs changing.

File: src/Inquirer/Validators/PercentageValidator.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace InquirerCore.Validators 
+{
+    public class PercentageValidator : IValidator
+    {
+        public bool Validate(string value)
+        {
+            try
+            {
+                var numberWithoutPercent = value.Replace("%", "");
```
Comment (afucher) [inline src/Inquirer/Validators/PercentageValidator.cs]: Reading the code I think that values like "75%%", "7%5" or "%75" will be valid, could you check this? 
Those values that I mentioned could be added to the tests data.

**Review by afucher — CHANGES_REQUESTED** [review]
Thanks for the changes, I just found some corner cases.

**Comment by Reisclef** [comment]
Alright, sure thing. Additional unit tests done. Also added an extra expected fail test for another symbol, in this case a dollar symbol.

**Comment by afucher** [comment]
Hey @Reisclef, sorry, I think that I didn't express myself clear. Those values that I mentioned, are supposed to fail.

Like, "75" and "75%" are a valid percentage value, but "7%5" and "75%%" are not.

**Comment by Reisclef** [comment]
Ah, I did wonder why that would be considered valid! Okay, so only a percentage at the end is valid. I'll look into making the appropriate changes.

**Comment by Reisclef** [comment]
New commit pushed with additional tests validating various inputs. Let me know if you have any further comments.

**Review by afucher — APPROVED** [review]
Nice, thanks for your contribution!
